### PR TITLE
ci: Run tests on all pull request

### PR DESCRIPTION
**Pull request by Gustl22**
# Description

If want to create a PR against a branch other than main, the tests are not executed. As there are very few of the mentioned PRs and it is more error prone, I propose to enable these, to better maintain other `release` branches.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `refactor:`,
      `docs:`, `chore:`, `test:`, `ci:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation and added dartdoc comments with `///`, where necessary.
- [x] I have updated/added relevant examples in [example].

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/bluefireteam/audioplayers/issues
[Contributor Guide]: https://github.com/bluefireteam/audioplayers/blob/main/contributing.md#feature-requests--prs
[Conventional Commit]: https://conventionalcommits.org
[example]: https://github.com/bluefireteam/audioplayers/tree/main/packages/audioplayers/example

### CODE REVIEW

**Review by spydon — APPROVED** [review]
I don't understand the last part, what do you mean with:

> to better maintain other release branches.

Other than that it looks good!

**Comment by Gustl22** [comment]
@spydon as we recently had difficulties distinguish breaking changes from the minor release, it may be a good approach to merge breaking changes in a separate branch instead of the `main` one. So fixes and smaller feature upgrade could still be released in the usual way while working on the next major `release branch` with breaking changes. Therefore Pull request have to be tested against the `release` branch and not the `main` branch. These `release` branches could always be rebased to the main branch until the actual major release, to avoid any conflicts or duplicate commits.

In summary of my approach: nothing changes, only breaking changes are hold back (collected in a release branch like `v6`) and merged just before the major release is desired.

**Comment by spydon** [comment]
@Gustl22 if we're going to continue following scaled trunk based development we should branch out from the last tag and then cherry pick changes in from `main` if there has been a breaking change done in the that we don't want to release yet, so almost the flow that you described except that the release branches never go back into main again.
If we have breaking changes that doesn't support this flow I think most of the time we should just release a new major release, to lower the maintenance burden.
https://trunkbaseddevelopment.com/branch-for-release/
